### PR TITLE
sagetex: init at 3.6

### PR DESCRIPTION
**Pull request by Alexnortung**
###### Motivation for this change

I want to add the sagetex latex package to nixpkgs. This package works and makes `sagetx.sty` know to latex, however since sage is not packaged as a python module, it is not possible to build the outputs of the `.sagetex.sage` file.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/misc/sagetex/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, pkgs, lib, ... }:
```
Comment (IvarWithoutBones) [inline pkgs/misc/sagetex/default.nix:1]: Individual derivations shouldn't reference `pkgs`, but instead the packages it needs.
```suggestion
{ lib
, stdenv
, fetchurl
}:
```

File: pkgs/misc/sagetex/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+stdenv.mkDerivation rec {
+  version = "3.6";
+  pname = "sagetex";
+  passthru.tlType = "run";
+
+  src = pkgs.fetchurl {
```
Comment (IvarWithoutBones) [inline pkgs/misc/sagetex/default.nix:8]:
```suggestion
  src = fetchurl {
```

File: pkgs/misc/sagetex/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    mkdir -p "$path"
+    cp -va *.sty *.cfg *.def "$path/"
+  '';
+
+  meta = with lib; {
+    description = "embed code, results of computations, and plots from Sage into LaTeX documents";
+    license = licenses.gpl2Plus;
+    maintainers = with maintainers; [
+      alexnortung
+    ];
```
Comment (IvarWithoutBones) [inline pkgs/misc/sagetex/default.nix:24]:
```suggestion
    maintainers = with maintainers; [ alexnortung ];
```
Looks a bit cleaner IMO

File: pkgs/misc/sagetex/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    sha256 = "fzol58VztvRLJHaaMYQ8gmMvSa5Lu2AZJ74buDPle4E=";
+  };
+
+  installPhase = ''
+    path="$out/tex/latex/sagetex"
+    mkdir -p "$path"
+    cp -va *.sty *.cfg *.def "$path/"
+  '';
+
+  meta = with lib; {
```
Comment (IvarWithoutBones) [inline pkgs/misc/sagetex/default.nix:19]: Please add the `meta.homepage` attribute.

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]
Optimally we'd build this from source, but I think this should be fine for now.

**Comment by collares** [comment]
I didn't try installing this package, but with Sage installed and `st_example.tex` from https://doc.sagemath.org/html/en/tutorial/sagetex.html, the following works if I have upstream's `sagetex.py` and `sagetex.sty` in the same directory as `st_example.tex`:

```
$ pdflatex st_example.tex
$ sage st_example.sagetex.sage
$ pdflatex st_example.tex
```

Judging by the "it is not possible to build the outputs of the .sagetex.sage file", I think we are using different workflows. What breaks in your case? I understand why importing sage from Python would be a good thing to support in general, though, and hopefully we'll be able to do that next year.

**Comment by Alexnortung** [comment]
@collares Yes I guess we have different workflows, I was getting an error that sage could not be imported. But that is another issue, and as you have tried it works as expected :smiley:
So I am looking forward for this to get merged.
However I am not sure if the package files are in the correct directory, but I hope it is fine.

**Comment by Alexnortung** [comment]
I have been using this for the past few days and works fine, so I think it is ready.

**Review by legendofmiracles — COMMENTED** [review]
Please squash all related commits

**Comment by Alexnortung** [comment]
The commits have now been squashed @legendofmiracles

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 151876` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sagetex</li>
  </ul>
</details>

File: pkgs/misc/sagetex/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, texlive
+}:
+
+stdenv.mkDerivation rec {
+  version = "3.6";
+  pname = "sagetex";
```
Comment (legendofmiracles) [inline pkgs/misc/sagetex/default.nix:9]:
```suggestion
  pname = "sagetex"; 
  version = "3.6";
```

**Comment by legendofmiracles** [comment]
Can you please fix the failing editorconfig check? 

After that LGTM, although I don't know how to test this

File: pkgs/misc/sagetex/default.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, texlive
+}:
+
+stdenv.mkDerivation rec {
+  pname = "sagetex";
```
Comment (collares) [inline pkgs/misc/sagetex/default.nix:8]:
```suggestion
  pname = "sagetex";
```

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 151876` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sagetex</li>
  </ul>
</details>

**Pull request merged**